server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -623,7 +698,43 @@ public int compare(File o1, File o2)
           if (Ints.tryParse(segmentDir.getName()) == null) {
             continue;
           }
-
+          QueryableIndex queryableIndex = null;
+          try {
+            queryableIndex = IndexIO.loadIndex(segmentDir);
```

Comment:
I don't see anywhere that this gets closed. It should be getting closed when the sink is dropped.

I may have missed something, but it's possible that it is buggy and nobody notices, because GC pressure is usually pretty high on realtime nodes and so the finalizers will unmap stuff in a somewhat timely manner.